src/main/java/software/amazon/awssdk/crt/CRT.java

```diff
@@ -300,7 +300,8 @@ private static void extractAndLoadLibrary(String path) {
             }
 
             // open a stream to read the shared lib contents from this JAR
-            String libResourcePath = "/" + os + "/" + getArchIdentifier() + "/" +  getCRuntime(os) + "/" + libraryName;
+            CrtPlatform platform = getPlatformImpl();
+            String libResourcePath = platform != null ? platform.getResourcePath(getCRuntime(os), libraryName) : "/" + os + "/" + getArchIdentifier() + "/" +  getCRuntime(os) + "/" + libraryName;
```

Comment:
When would platform be null?

Reply:
if `findPlatformImpl()` doesn't find CrtPlatformImpl in an OS specified location it will be null. As of right now, Android should be the only platform that implements it outside of testing environments.

Reply:
Well, I'm not a fan of the fact that the default path is implemented twice.

Reply:
Changed it. It's a little messier but there's only a singular default path.

Reply:
just a note: I looked into some way of combining and using different ways to use existing functions but the problem was that android does so many very android specific different things in terms of `getOSIdentifier()` for this path, `getArchIdentifier()`, and omitting `getCRuntime()` entirely that it felt best to add this one filepath function in one place rather than peppering android specific things across `CRT.java` and/or introducing multiple platform specific functions and branches to OS/Arc/Runtime identifiers. Open to suggestions on it tho if you have any ideas.